### DISCUSSION TITLE
FilRep.io missing log entries for miner reachability query

### DISCUSSION THREAD

**Discussion by shotcollin** [filecoin-project/lotus#8693] [Storage Provider]
Does anyone know if [FilRep.io](https://filrep.io/) is actively maintained? I noticed it hasn't updated info for my miner for about a month now, and querying at random a few of the listed miners I found that several listed as unreachable are definitely reachable and several listed as reachable are currently offline. Many also didn't have accurate price info listed in the FilRep registry. 

The only contact listed on the site is an email address, contact@filrep.io, but mail to that address (tested with both gmail and non-gmail sender) results in an immediate response from `mailer-daemon@googlemail.com` saying `We're writing to let you know that the group you tried to contact (contact) may not exist, or you may not have permission to post messages to the group...`.

Seems like FilRep properly updating data points would be important given that the [Filecoin Plus Registry](https://plus.fil.org/miners) appears to be using FilRep scores and possibly other data.

**Comment by kiriakos** [comment]
Hi @shotcollin, did you ever get a response to this?

**Comment by shotcollin** [comment]
Hi @kiriakos – I didn't get any response, no. 

I'm not sure the page is working at all anymore, actually. It doesn't seem to load new results and when I tried to log in a minute ago it just said "internal server error".

**Reply by cezarparaschiv** [reply]
Hey, this is Cezar from filrep.io. I just came across this. Filrep is currently being maintained. In the past, we experienced a significant backlog of unprocessed data in the backend. Are you still having any issues?

**Reply by shotcollin** [reply]
Hi @cezarparaschiv. Sorry for they delayed response. I'm still seeing issues. Main issue is that logging in with a previously-created account returns a 500 Internal Server Error. Same thing with attempting to create a new account. This is the case on several different browsers tried on a couple different systems. 

Also I noticed that of the first 300 SPs polled, 155 of them are reported to be "unreachable". And after that, almost all of them are marked unreachable. Is it really the case that out of 6,300 storage providers listed there are actually only a couple hundred online? Is that also related to the backlog of unprocessed data?

**Reply by cezarparaschiv** [reply]
@shotcollin the account creation is fixed. Can you please check? Related to the accuracy of the "reachable/unreachable" data, the bot seems to work just fine, we haven't noticed any glitch in how it operates.

**Reply by shotcollin** [reply]
Hi @cezarparaschiv, thanks for looking into this. I was able to log in just now. The site seems pretty slow to respond, but I was able to look again at the logs for my miner. 

In the time since I originally looked at these in May 2022, it does appear the bot made some additional queries to the miner, but they're all still reporting a status of "failed". 

What's strange is that the miner has been running this entire time (albeit with some very limited periods offline), and also there are some huge gaps in the logs. They start in Sept. 2021, shortly after the miner was initialized. They continue through almost the end of Oct. 2021, then stop. There are three log entries on two days in late November, then the next is 2022-04-10. After that, there's one entry on 2022-04-18, and then the single status of "ok"  was returned on 2022-04-21 (out of 4,489 queries). The next log entry is almost a year later, 2023-04-04. Since then the entries have been fairly consistent, although they skip most of Aug. and Sept. 2023, and then the last one is 2023-11-16. 

I'm wondering if you can provide some insight as to what's being queried exactly, and why the miner would have a failed status for all the queries? I've looked at the methodology section on the site, but I'm still unclear what the bot is querying—is it just a heartbeat test that determines reachability, or is it querying the FIL price for storage deals? And do the gaps represent other times the miner was unreachable that just weren't logged, or is there some other explanation for the large gaps with no log entries at all? 

The miner id is f01197354 if that helps. Thanks again for your help with this.

**Reply by cezarparaschiv** [reply]
@shotcollin, the site is indeed experiencing slow performance. We have identified the issue and implemented a temporary fix. Currently, our team is actively working on a permanent solution. Also, you are correct about the bot glitches; it seems to be related to a RAM issue affecting its optimal performance. We are working to resolve this matter as well.

**Reply by shotcollin** [reply]
Thanks for the info @cezarparaschiv. Does this mean there's a glitch with the bot that's causing it to not log queries made to certain miners for long periods of time? Or is it that the bot isn't making the queries at all? 

Also can you provide any insight or even speculation as to why the bot might have only reported an ok status once in ~4,500 queries, when bot was logging queries to made to f01197354? Could it be a miner configuration issue? Or something else? I know the miner was correctly configured to accept deals and was reachable *at some point* during the period the bot was reporting it wasn't, because there were multiple deals made that weren't my own testing. I wouldn't rule out misconfiguration for that entire period though, so it might help to know what exactly the bot is querying for when it's doing its query for reachability, in order to figure out if it was on my end.

To that same point, it seems there are at least 150-200 miners the bot has been able to successfully query without interruption. Have you been able to identify any significant differences between the those miners and any potentially affected by the bot glitch that's causing queries to not be logged and/or made?

**Reply by cezarparaschiv** [reply]
1.Does this mean there’s a glitch with the bot that’s causing it to not log queries made to certain miners for long periods of time?
   Or is it that the bot isn’t making the queries at all?
**When the glitch happened the bot wasn't making queries at all**
2. Also can you provide any insight or even speculation as to why the bot might have only reported an ok status once in ~4,500 queries,when bot was logging queries to made to f01197354?
Could it be a miner configuration issue? Or something else?
I know the miner was correctly configured to accept deals and was reachable at some point during the period the bot was reporting it wasn’t, because there were multiple deals made that weren’t my own testing.
I wouldn’t rule out misconfiguration for that entire period though, so it might help to know what exactly the bot is querying for when it’s doing its query for reachability, in order to figure out if it was on my end.
**The ok status means the miner is responding to ClientQueryAsk request with a price, if is not configure to accept deals the call to ClientQueryAsk will fail**
3. To that same point, it seems there are at least 150-200 miners the bot has been able to successfully query without interruption.
	Have you been able to identify any significant differences between the those miners and any potentially affected by the bot glitch that’s causing queries to not be logged and/or made?
**During the glitch no queries ware made so for each miner the dashboard would display the last known state of the miner.**
